src/meta/processors/admin/HBProcessor.cpp

```diff
@@ -37,25 +37,32 @@ void HBProcessor::process(const cpp2::HBReq& req) {
         return;
     }
 
-    LOG(INFO) << "Receive heartbeat from " << host;
     auto ret = kvstore::ResultCode::SUCCEEDED;
+    bool updateLeader = false;
+    int64_t lastUpdateTime;
     if (req.get_in_storaged()) {
+        LOG(INFO) << "Receive heartbeat from " << host;
         HostInfo info(time::WallClock::fastNowInMilliSec());
-        ret = ActiveHostsMan::updateHostInfo(kvstore_, host, info);
         if (req.__isset.leader_partIds) {
-            const auto& leaderIds = req.get_leader_partIds();
-            UNUSED(leaderIds);
-            // TODO(zhangguoqing) put the leader parts information into kvstore
+            ret = ActiveHostsMan::updateHostInfo(kvstore_, host, info,
+                                                 req.get_leader_partIds());
+            updateLeader = true;
+            lastUpdateTime = time::WallClock::fastNowInMilliSec();
+            LastUpdateTimeMan::update(kvstore_, lastUpdateTime);
```

Comment:
It seems that we don't need to update the lastUpdateTime in HeartBeat processor.

Reply:
Indeed, fixed